.github/workflows/build.yml

```diff
@@ -19,17 +20,147 @@ jobs:
           toolchain: nightly-2021-05-11
           override: true
           target: wasm32-unknown-unknown
-    - name: Install llvm-10
-      run: sudo apt-get install llvm-10 clang-10
+    - name: Install System Level Dependencies
+      run: sudo apt-get install build-essential ocaml ocamlbuild automake autoconf libtool wget python libssl-dev git cmake perl pkg-config curl llvm-10 clang-10 libclang-10-dev
     - uses: actions/checkout@v2
       with:
         submodules: 'true'
-    - name: Build
+        
+    - name: Build core blockchain
       run: cargo build --verbose --release
-    - name: Save binaries
-      uses: actions/upload-artifact@v1
+    - name: Save core-blockchain binaries
+      uses: actions/upload-artifact@v2
       with:
-        name: binaries
-        path: target/release/phala-node
-    - name: Tests
+        name: core-blockchain-binaries
+        path: ./target/release/phala-node
+    - name: Save pherry binaries
+      uses: actions/upload-artifact@v2
+      with:
+        name: pherry-binaries
+        path: ./target/release/pherry
+        
+  build-pruntime:
+    name: build-pruntime
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Install latest nightly
+      uses: actions-rs/toolchain@v1
+      with:
+          toolchain: nightly-2021-05-11
+          override: true
+          target: wasm32-unknown-unknown
+    - name: Install System Level Dependencies
+      run: sudo apt-get install build-essential ocaml ocamlbuild automake autoconf libtool wget python libssl-dev git cmake perl pkg-config curl llvm-10 clang-10 libclang-10-dev
+    - name: Download Intel SGX SDK
+      run: wget https://download.01.org/intel-sgx/sgx-linux/2.12/distro/ubuntu20.04-server/sgx_linux_x64_sdk_2.12.100.3.bin
+    - name: Install Intel SGX SDK
+      run: chmod +x ./sgx_linux_x64_sdk_2.12.100.3.bin && echo -e 'no\n/opt/intel' | sudo ./sgx_linux_x64_sdk_2.12.100.3.bin
+    - name: Setup Node.js environment
+      uses: actions/setup-node@v2.4.0
+      with:
+        node-version: 14.x
+    - name: Install yarn 2
+      run: sudo npm install -g yarn && yarn set version berry
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'true'
```

Comment:
I think node.js is not necessary in building pRuntime. Did you introduce it for caching between the actions?

Reply:
My idea is that later we can composite the dependency installing process (so we can reuse it for better maintenance), so for now basically all jobs share the same dependencies.

Reply:
I think you are right. Removing nodejs installing in `pruntime` compiling.